workflow/lib/preprocess/preprocess.py

```diff
@@ -1,61 +1,114 @@
     """
+
     all_metadata = []
-    
+
     # Iterate through all provided files
     for file_path in files:
-        if file_path.endswith('.nd2'):
-            tile = extract_tile_from_filename(file_path)
-            
-            with ND2Reader(file_path) as images:
-                raw_metadata = images.parser._raw_metadata
-                
-                data = {
-                    'x_data': raw_metadata.x_data,
-                    'y_data': raw_metadata.y_data,
-                    'z_data': images.metadata['z_coordinates'],
-                    'pfs_offset': raw_metadata.pfs_offset,
-                    'field_of_view': tile,
-                    'filename': file_path,
-                }
-                
-                df = pd.DataFrame(data)
-                
-                if 'z_levels' in images.metadata and set(images.metadata['z_levels']) == set(range(0, 4)):
-                    df = df.iloc[::4, :]
-                all_metadata.append(df)
-    
+        tile = extract_tile_from_filename(file_path)
+
+        with ND2Reader(file_path) as images:
+            raw_metadata = images.parser._raw_metadata
+
+            data = {
+                "x_data": raw_metadata.x_data,
+                "y_data": raw_metadata.y_data,
+                "z_data": images.metadata["z_coordinates"],
+                "pfs_offset": raw_metadata.pfs_offset,
+                "field_of_view": tile,
+                "filename": file_path,
+            }
+
+            df = pd.DataFrame(data)
+
+            if "z_levels" in images.metadata and set(
+                images.metadata["z_levels"]
+            ) == set(range(0, 4)):
+                df = df.iloc[::4, :]
+            all_metadata.append(df)
+
     # Combine all metadata
     if all_metadata:
         combined_metadata = pd.concat(all_metadata, ignore_index=True)
-        
+
         # Convert 'field_of_view' to numeric, coercing any non-numeric values to NaN
-        combined_metadata['field_of_view'] = pd.to_numeric(combined_metadata['field_of_view'], errors='coerce')
+        combined_metadata["field_of_view"] = pd.to_numeric(
+            combined_metadata["field_of_view"], errors="coerce"
+        )
         # Sort by 'field_of_view' in ascending order
-        combined_metadata = combined_metadata.sort_values('field_of_view').reset_index(drop=True)
+        combined_metadata = combined_metadata.sort_values("field_of_view").reset_index(
+            drop=True
+        )
         return combined_metadata
     else:
-        print(f"No valid ND2 files found in the provided list.")
+        print("No valid ND2 files found in the provided list.")
         return pd.DataFrame()  # Return an empty DataFrame if no files were processed
+
+
+def nd2_to_tif(file: str, channel_order_flip: bool = False) -> np.ndarray:
+    """
+    Converts a single ND2 file with one field of view and multiple channels to a multidimensional numpy array.
```

Comment:
So as we migrate, are we assuming this format? Are we ignoring the potential to use well based images?

Reply:
By format are you referring to the single ND2 file with one field of view and multiple channels? I'm not sure if we should assume this format and ignore the potential to use well-based images. Let's discuss over Zoom before we come to a conclusion on this.